### PR TITLE
Introduce zero address check to `IsValidAddress`

### PR DESCRIPTION
**Pull request by dusannosovic-ethernal**
# Description

This PR include additional check for zeroAddress in `types.IsValidAddress`

# Changes include

- [X] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: types/types.go [hunk 1]
```diff
@@ -186,6 +186,10 @@ func IsValidAddress(address string) error {
 		return fmt.Errorf("address %s has invalid length", string(decodedAddress))
 	}
 
+	if addr := StringToAddress(address); addr == ZeroAddress && !zeroAddressAllowed {
```
Comment (Stefan-Ethernal) [inline types/types.go]: I'd suggest returning the parsed address since we are parsing it already, in case it is valid.

**Review by goran-ethernal — DISMISSED** [review]

File: types/types.go [hunk 2]
```diff
@@ -186,6 +186,10 @@ func IsValidAddress(address string) error {
...
+		return fmt.Errorf("address can't be zeroAddress")
```
Comment (Stefan-Ethernal) [inline types/types.go]:
```suggestion
		return errors.New("zero address is not allowed")
```

File: types/types.go (hunk 1 above)
Comment (Stefan-Ethernal) [reply types/types.go]: Fixed https://github.com/Ethernal-Tech/blade/pull/56/commits/f21a17a9c3b56e71fc50ef0dcf3665cfb67a6b78

File: command/genesis/params.go [hunk 3]
```diff
@@ -190,15 +190,12 @@ func (p *genesisParams) validateFlags() error {
 			return err
 		}
 
-		if err := types.IsValidAddress(params.stakeToken); err != nil {
+		stakeTokenAddr, err := types.IsValidAddress(params.stakeToken, false)
```
Comment (dusannosovic-ethernal) [inline command/genesis/params.go]:
```suggestion
		params.stakeTokenAddr, err := types.IsValidAddress(params.stakeToken, false)
```

File: command/validator/registration/params.go [hunk 4]
```diff
@@ -34,15 +34,12 @@ func (rp *registerParams) validateFlags() (err error) {
 		return fmt.Errorf("provided value (%d) is less than zero", rp.amountValue)
 	}
 
-	if err := types.IsValidAddress(params.stakeToken); err != nil {
+	stakeTokenAddr, err := types.IsValidAddress(params.stakeToken, false)
```
Comment (dusannosovic-ethernal) [inline command/validator/registration/params.go]:
```suggestion
	rp.stakeTokenAddr, err := types.IsValidAddress(params.stakeToken, false)
```

File: command/validator/registration/params.go [hunk 5]
```diff
@@ -34,15 +34,12 @@ func (rp *registerParams) validateFlags() (err error) {
...
+	if err != nil {
 		return fmt.Errorf("stake token address is not a valid address: %w", err)
 	}
 
-	params.stakeTokenAddr = types.StringToAddress(params.stakeToken)
-
-	if params.stakeTokenAddr == types.ZeroAddress {
-		return errStakeTokenIsZeroAddress
-	}
+	params.stakeTokenAddr = stakeTokenAddr
```
Comment (dusannosovic-ethernal) [inline command/validator/registration/params.go:42]:
```suggestion

```

File: command/validator/stake/params.go [hunk 6]
```diff
@@ -36,15 +36,12 @@ func (sp *stakeParams) validateFlags() (err error) {
 		return fmt.Errorf("failed to parse json rpc address. Error: %w", err)
 	}
 
-	if err := types.IsValidAddress(sp.stakeToken); err != nil {
+	stakeTokenAddr, err := types.IsValidAddress(sp.stakeToken, false)
```
Comment (dusannosovic-ethernal) [inline command/validator/stake/params.go]:
```suggestion
	sp.stakeTokenAddr, err := types.IsValidAddress(sp.stakeToken, false)
```

File: command/genesis/params.go (hunk 3 above)
Comment (Stefan-Ethernal) [reply command/genesis/params.go]: https://github.com/Ethernal-Tech/blade/pull/56/commits/c6e980793bea2858b0ed43e332a4628f65f48043

File: command/validator/registration/params.go (hunk 4 above)
Comment (Stefan-Ethernal) [reply command/validator/registration/params.go]: https://github.com/Ethernal-Tech/blade/pull/56/commits/c6e980793bea2858b0ed43e332a4628f65f48043

File: command/validator/stake/params.go (hunk 6 above)
Comment (Stefan-Ethernal) [reply command/validator/stake/params.go]: https://github.com/Ethernal-Tech/blade/pull/56/commits/c6e980793bea2858b0ed43e332a4628f65f48043

**Review by Stefan-Ethernal — DISMISSED** [review]

File: command/bridge/premine/params.go [hunk 7]
```diff
@@ -13,51 +12,37 @@ import (
...
+	accountDir      string
+	accountConfig   string
+	privateKey      string
+	bladeManager    string
+	nativeTokenRoot string
+	jsonRPC         string
+	stakedAmount    string
+	premineAmount   string
+
+	nonStakedValue      *big.Int
```
Comment (goran-ethernal) [inline command/bridge/premine/params.go]:
```suggestion
	premineAmoountValue      *big.Int
```

Comment (Stefan-Ethernal) [reply command/bridge/premine/params.go]: Fixed https://github.com/Ethernal-Tech/blade/pull/56/commits/3c92d8d5d13064173b67140661fb02fa54726835

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**